### PR TITLE
Run3-hcx361 Make Era dependent ZDC geometry - backport #45511

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Make Era dependent ZDC geometry - backport #45511

#### PR validation:

Use the runTheMatrix test workflows 

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
backport of https://github.com/cms-sw/cmssw/pull/45511

**Comment by mandrenguyen** [comment]
ZDV = ZDC ?

**Comment by mandrenguyen** [comment]
urgent
Needed to build 14_1_0 for heavy-ion data taking

**Comment by bsunanda** [comment]
@aloeliger, @civanch, @consuegs, @epalencia, @francescobrivio,  @mmusich, @perrotta, @saumyaphor4252, @srimanob Please approve this back port

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -261,6 +261,18 @@ def checkHLTfor43774(process):
 
     return process
 
+# Adding zdc Topology producer
+def customizeHLTfor45511(process):
```
Comment (mmusich) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:265]:
```suggestion
def customizeHLTfor46011(process):
```

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 2]
```diff
@@ -327,5 +339,6 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
     process = checkHLTfor43774(process)
     process = customizeHLTfor44576(process)
     process = customizeHLTfor45063(process)
-
+    process = customizeHLTfor45511(process)
```
Comment (mmusich) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:342]:
```suggestion
    process = customizeHLTfor46011(process)
```

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by mmusich** [comment]
+hlt

**Comment by perrotta** [comment]
+1
- "Almost" verbatim backport of the PR in the master, with identical functionalities

**Comment by bsunanda** [comment]
@aloeliger, @civanch, @consuegs, @epalencia, @srimanob Please approve this PR - it is urgent

**Comment by mmusich** [comment]
can this backport include directly also the commits of https://github.com/cms-sw/cmssw/pull/46033 ?

**Comment by srimanob** [comment]
+Upgrade

**Comment by bsunanda** [comment]
+geometry

**Comment by bsunanda** [comment]
@aloeliger, @civanch, @consuegs, @epalencia Please approve this PR

**Comment by bsunanda** [comment]
#46033 is still not approved and it can be backported once this PR is merged. I urgently seek merging of this PR

**Comment by civanch** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Comment by mandrenguyen** [comment]
merge 
bypassing L1 in the interest of time, but they signed the master version

**Comment by aloeliger** [comment]
Apologies, but that's fine by us.

**Comment by abdoulline** [comment]
Now the RelVal results are not available anymore,  
the question arises - why so many differences were reported in the tests... 
https://github.com/cms-sw/cmssw/pull/46011#issuecomment-2357059478

**Comment by kpedro88** [comment]
@abdoulline changes in the SIM-level geometry lead to changes in random number calls, resulting in statistical fluctuations.

**Comment by abdoulline** [comment]
> @abdoulline changes in the SIM-level geometry lead to changes in random number calls, resulting in statistical fluctuations.

I thought it's about preparatory work, not yet any Geometry changes deployed... 🤔
If I remember it right, all the changes came at DIGI level (because of HcalDigitizer.cc modifications?) and I've even answered in a similar way to Andrea Perrotta some time ago in another PR chat, when was asked about massive differences. 
But I'm just not sure anymore if there were only these random changes...

**Pull request merged**